community/modules/scheduler/schedmd-slurm-gcp-v6-controller/controller.tf

```diff
@@ -123,7 +123,27 @@ resource "google_secret_manager_secret" "cloudsql" {
   secret_id = "${local.slurm_cluster_name}-slurm-secret-cloudsql"
 
   replication {
-    auto {}
+    dynamic "auto" {
+      for_each = length(var.cloudsql.user_managed_replication) == 0 ? [1] : []
```

Comment:
Don't you need to check if `var.cloudsql.user_managed_replication != null` ?

Reply:
@tpdownes as it comes from your code :-) WDYT?

Reply:
I have added a code suggestion that aligns the default value with the one used above (empty list, not null).

https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/2828/files#r1698997872

If someone goes out of their way to set this to null, that's not something I think we need to handle.